helpdesk-ai/src/main/java/org/symphonyoss/symphony/bots/ai/message/MessageProducer.java

```diff
@@ -26,24 +38,29 @@
 
   private final UsersClient usersClient;
 
+  private final SymphonyClientUtil symphonyClientUtil;
+
   public MessageProducer(MessagesClient messagesClient, MembershipClient membershipClient,
-      UsersClient usersClient) {
+      UsersClient usersClient, SymphonyClient symphonyClient) {
```

Comment:
I believe you don't require messagesClient and usersClient once you have already the symphonyClient as a parameter. I mean you can extract these clients from symphonyClient.

Example:
```
public MessageProducer(SymphonyClient symphonyClient, MembershipClient membershipClient) {
  this.messagesClient = symphonyClient.getMessagesClient();
  this.usersClient = symphonyClient.getUsersClient();
  this.membershipClient = membershipClient;
  this.symphonyClientUtil = new SymphonyClientUtil(symphonyClient);
}
```